### ISSUE TITLE
Different arg names for conv1d

### ISSUE THREAD

**Issue by evan176** [tensorflow/tensorflow#10311]
### System information
- **TensorFlow version (use command below)**: r1.1

### Describe the problem
In latest verion, tf.nn.conv1d and tf.nn.conv2d has different arg names for filter and stride.
tf.nn.conv1d: 
    filters: A 3D Tensor. Must have the same type as input.
    stride: An integer. The number of entries by which the filter is moved right at each step.
tf.nn.conv2d:
    filter: A Tensor. Must have the same type as input. A 4-D tensor of shape [filter_height, filter_width, in_channels, out_channels]
    strides: A list of ints. 1-D tensor of length 4. The stride of the sliding window for each dimension of input. The dimension order is determined by the value of data_format, see below for details.

Should it be unified to same name?

**Comment by Androbin** [comment]
The naming for conv2d and conv3d seems to match: `input, filter, strides, padding`
This should be used in conv1d as well: `value -> input, filters -> filter, stride -> strides`

**Comment by asimshankar** [comment]
Ah, this discrepancy is unfortunate. Thanks for pointing it out.

Alas, since this now part of the 1.x release - we can't get rid of the old names. However, we can perhaps add "input", "filter" and "strides" as arguments to conv1d.

I'll take a stab at that.

(Though, perhaps `stride` still makes sense since it is a single integer, not a list as in `conv2d` and `conv3d`?)

**Comment by asimshankar** [comment]
Actually, on further reflection - while this discrepancy is unfortunate - I don't think it is worth fixing.

We have to maintain backward compatibility of the API for all 1.x releases. In order to accept either the old or new arguments as named parameters, the signature of `conv1d` will have to change from:

```
 def conv1d(value, filters, stride, padding,                                                                                                  
           use_cudnn_on_gpu=None, data_format=None,
           name=None)
```

to something like:

```
def conv1d(value=None, filters=None, stride=None, padding=None,                                                                                                  
           use_cudnn_on_gpu=None, data_format=None,
           name=None, input=None, filter=None, strides=None)
```

With a bunch of logic in the new function to ensure that:
- At least one of `input` or `value` is not `None`
- Same for one of `filter` or `filters` and one of `stride` or `strides`
- `padding` is not `None`
- If `value` is specified, prefer that over `input`
etc.

I fear that doing so will add more confusion than it helps resolve.

Given that, I'm proposing that we live with this unfortunate historical artifact.

Let me know if anyone feels particularly strongly, otherwise I'll go ahead and close this issue in a day or two.

**Comment by foxik** [comment]
+1 to not fixing the issue

**Comment by Androbin** [comment]
We should probably at least keep a list of such planned API changes for 2.x

**Comment by tensorflowbutler** [comment]
This issue is automatically closed due to lack of activity. Please re-open if this is still an issue for you. Thanks!
